apps/passport-server/test/generic-issuance/pipelines/auto-issuance/autoIssuanceSpec.ts

```diff
@@ -0,0 +1,1089 @@
+    giService = giBackend.services
+      .genericIssuanceService as GenericIssuanceService;
+    await giService.stop();
+    const pipelineDefinitionDB = new PipelineDefinitionDB(
+      giBackend.context.dbPool
+    );
+    await pipelineDefinitionDB.deleteAllDefinitions();
+    await pipelineDefinitionDB.upsertDefinitions(pipelineDefinitions);
+    await giService.start(false);
+  });
+
+  this.beforeEach(async () => {
+    MockDate.set(now);
+  });
+
+  this.afterEach(async () => {
+    mockServer.resetHandlers();
+    MockDate.reset();
+  });
+
+  this.afterAll(async () => {
+    mockServer.close();
+  });
+
+  step("PipelineUserDB", async function () {
+    const userDB = new PipelineUserDB(giBackend.context.dbPool);
+
+    const adminUser: PipelineUser = {
+      id: adminGIUserId,
+      email: adminGIUserEmail,
+      isAdmin: true,
+      timeCreated: nowDate,
+      timeUpdated: nowDate
+    };
+    await userDB.updateUserById(adminUser);
+    assertUserMatches(
+      {
+        id: adminGIUserId,
+        email: adminGIUserEmail,
+        isAdmin: true,
+        timeCreated: nowDate,
+        timeUpdated: nowDate
+      },
+      await userDB.getUserById(adminUser.id)
+    );
+
+    // TODO: comprehensive tests of create update read delete
+  });
+
+  /**
+   * Test for {@link PretixPipeline} for Auto Issuance event.
+   */
+  step(
+    "PretixPipeline issuance and checkin and PipelineInfo for Auto Issuance",
+    async () => {
+      expectToExist(giService);
+      const pipelines = await giService.getAllPipelineInstances();
+      expectToExist(pipelines);
+      expectLength(pipelines, 1);
+      const pipeline = pipelines.find(PretixPipeline.is);
+      expectToExist(pipeline);
+      expect(pipeline.id).to.eq(autoIssuancePipeline.id);
+      const autoIssuanceTicketFeedUrl = pipeline.issuanceCapability.feedUrl;
+      const autoIssuanceIssuanceDateTime = new Date();
+
+      const foodVendorTickets = await requestTicketsFromPipeline(
+        pipeline.issuanceCapability.options.feedFolder,
+        autoIssuanceTicketFeedUrl,
+        pipeline.issuanceCapability.options.feedId,
+        ZUPASS_EDDSA_PRIVATE_KEY,
+        pretixBackend.get().autoIssuanceOrganizer.autoIssuanceFoodVendorEmail,
+        AutoIssuanceAttendeeIdentity
+      );
+      expectLength(foodVendorTickets, 2); // just their food vendor ticket and corresponding pod ticket
+
+      const attendeeTickets = await requestTicketsFromPipeline(
+        pipeline.issuanceCapability.options.feedFolder,
+        autoIssuanceTicketFeedUrl,
+        pipeline.issuanceCapability.options.feedId,
+        ZUPASS_EDDSA_PRIVATE_KEY,
+        pretixBackend.get().autoIssuanceOrganizer.autoIssuanceAttendeeEmail,
+        AutoIssuanceAttendeeIdentity
+      );
+      expectLength(
+        attendeeTickets.map((t) => t.claim.ticket),
+        (1 + 2) * 2 // 1 ticket + pod + 2 vouchers + 2 voucher pods
+      );
+
+      {
```

Comment:
food voucher issuance tests in this block
- you get the right amount of food vouchers
- per bucket unit of time
- you don't get more food vouchers after auto-issuance configured end date